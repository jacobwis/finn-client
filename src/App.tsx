import { hot } from 'react-hot-loader';
import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { CurrentUserQuery, CURRENT_USER_QUERY } from './api';
import * as AppContext from './contexts/AppContext';
import MainLayout from './components/MainLayout';

// views
import BookView from './views/BookView';
import HomeView from './views/HomeView';
import Playground from './Playground';
import SearchView from './views/SearchView';

class App extends React.Component {
  public render() {
    return (
      <CurrentUserQuery query={CURRENT_USER_QUERY}>
        {({ data }) => (
          <AppContext.Provider currentUser={data.currentUser}>
            <Switch>
              <Route path="/playground" component={Playground} />
              <Route
                render={() => (
                  <MainLayout>
                    <Route path="/" exact component={HomeView} />
                    <Route path="/search" component={SearchView} />
                    <Route
                      path="/books"
                      render={({ match }) => (
                        <>
                          <Route path={`${match.url}/:id`} component={BookView} />
                        </>
                      )}
                    />
                  </MainLayout>
                )}
              />
            </Switch>
          </AppContext.Provider>
        )}
      </CurrentUserQuery>
    );
  }
}

export default hot(module)(App);
