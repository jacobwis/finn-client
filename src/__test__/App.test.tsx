import * as React from 'react';
import { mount, shallow } from 'enzyme';
import App from '../App';

describe('App', () => {
  it('should run tests', () => {
    const wrap = shallow(<App />);
    expect(wrap).toMatchSnapshot();
  });
});
