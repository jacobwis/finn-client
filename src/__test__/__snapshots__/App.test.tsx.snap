// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should run tests 1`] = `
<div>
  <h1>
    Hello reloaded
  </h1>
</div>
`;
